lib/models/post_asset.ts

```diff
@@ -19,7 +19,8 @@ export = (ctx: Hexo) => {
     // PostAsset.path is file path relative to `public_dir`
     // no need to urlescape, #1562
     // strip /\.html?$/ extensions on permalink, #2134
-    return join(post.path.replace(/\.html?$/, ''), this.slug);
+    // Replace backslashes on Windows using posix.join
```

Comment:
posix.join will not replace backslash.

```node
path.posix.join('asd\\zxc','qwe')
```

Reply:
This comment should be changed to:
```
Use `path.posix.join` to avoid `path.join` introducing unwanted backslashes on Windows.
```
The input parameters here actually do not include backslashes, so the code is fine.